envisage/tests/test_extension_registry_mixin.py

```diff
@@ -0,0 +1,117 @@
+# (C) Copyright 2007-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+"""
+Base set of tests for extension registry and its subclasses wrapped in a
+mixin class.
+"""
+
+# Enthought library imports.
+from envisage.api import ExtensionPoint
+from envisage.api import UnknownExtensionPoint
+from traits.api import List
+
+
+class ExtensionRegistryTestMixin:
+    """ Base set of tests for extension registry and its subclasses. """
+
+    def setUp(self):
+        """ Prepares the test fixture before each test method is called. Test
+        cases inherriting from this mixin should override this method and
+        define self.registry.
+        """
+        pass
```

Comment:
Since this setup is empty, it might be better not to define it, so the mixin can be seen as contributing test methods, no one needs to to worry about `setUp`/`tearDown` being overridden by multi-inheritance. 

The requirement on `self.registry` should be documented though... maybe in the class docstring. And we should say what the type is (I think it is `ExtensionPointRegistry`?